LessonOneFinalModuleHW/Program.cs

```diff
@@ -0,0 +1,245 @@
+                string finishedTaskArrayEmpty = Array.Find(finishedTaskArray, i => i.StartsWith(value, StringComparison.Ordinal));
+
+                if (taskArrayEmpty != null && finishedTaskArrayEmpty == null)
+                {
+                    taskArray = taskArray.Where(i => i != value).ToArray();
+                }
+                else if (finishedTaskArrayEmpty != null && taskArrayEmpty == null)
+                {
+                    int indexDate = Array.FindIndex(finishedTaskArray, i => i.StartsWith(value, StringComparison.Ordinal));
+                    dateArray = dateArray.Where((source, index) => index != indexDate).ToArray();
+                    finishedTaskArray = finishedTaskArray.Where(i => i != value).ToArray();
+                    Console.WriteLine($"You're task-item '{value}' was deleted!");
+                }
+                else
+                {
+                    Console.WriteLine("You're task-item is not there!");
+                }
+            }
+
+            return (taskArray, finishedTaskArray, dateArray);
+        }
+
+        static void PrintArray(string[] taskArray, string[] finishedTaskArray, string[] dateArray, string status)
+        {
+            if (status == "1")
+            {
+                Console.WriteLine("[Finished]");
+                Console.WriteLine(" {0}", string.Join(" \n  ", finishedTaskArray));
+                Console.WriteLine("[Date]");
+                Console.WriteLine(" {0}", string.Join(" \n  ", dateArray));
+            } 
+            else if (status == "0")
+            {
+                Console.WriteLine("[Active]");
+                Console.WriteLine(" {0} ", string.Join(" \n ", taskArray));
+
+            }
+            else if (status == String.Empty)
+            {
+                Console.WriteLine("[Active]");
+                Console.WriteLine(" {0} ", string.Join(" \n  ", taskArray));
+                Console.WriteLine("[Finished]");
+                Console.WriteLine(" {0} ", string.Join(" \n  ", finishedTaskArray));
+                Console.WriteLine("[Date]");
+                Console.WriteLine(" {0} ", string.Join(" \n  ", dateArray));
+            }
+            else
+            {
+                Console.WriteLine("You're command input isn't valid");
+            }
+        }
+
+        static (string[] taskArray, string[] finishedTaskArray, string[] dateArray) MarkAsFunc(string[] taskArray, string[] finishedTaskArray, string[] dateArray, string value, string status)
+        {
+
+            if (value == String.Empty)
+            {
+                Console.WriteLine("You're task-item is empty!");
+            }
+
+            else
+            {
+                if (int.TryParse(status, out int statusResult))
+                {
+                    if (statusResult == 0) // task is not finished
+                    {
+                        string valueEmpty = Array.Find(finishedTaskArray, i => i.StartsWith(value, StringComparison.Ordinal));
```

Comment:
You are doing search operation 2+ times. Isn't it easier to find it once, return the index and use that index where needed?